radix-engine-interface/src/blueprints/consensus_manager/invocations.rs

```diff
@@ -105,16 +105,67 @@ pub struct EpochChangeCondition {
     pub target_duration_millis: u64,
 }
 
+pub enum EpochChangeOutcome {
+    NoChange,
+    Change {
+        next_epoch_effective_start_millis: i64,
+    },
+}
+
 impl EpochChangeCondition {
     /// Determines whether this condition is met by the given actual state.
     /// See the condition's field definitions for exact rules.
-    pub fn is_met(&self, duration_millis: i64, round: Round) -> bool {
+    pub fn should_epoch_change(
+        &self,
+        effective_start: i64,
+        current_time: i64,
```

Comment:
What's the behaviour if a proposer sets `current_time` to a negative value?

Reply:
Have added a new assertion and test case to ensure that the current time is non-decreasing.